addons/l10n_dk_bookkeeping/__init__.py

```diff
@@ -1,3 +1,10 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from . import models
+from odoo.tools import column_exists, create_column
+
+
+def pre_init_hook(env):
+    """Do not compute the l10n_dk_currency_rate_at_transaction field on existing Moves."""
+    if not column_exists(env.cr, "account_move", "l10n_dk_currency_rate_at_transaction"):
+        create_column(env.cr, "account_move", "l10n_dk_currency_rate_at_transaction", "numeric")
```

Comment:
So in this case, the column is created but you don't run the compute therefore it won't have a value. Not ideal. I see that in 17.3, we merged a PR that made this whole module obsolete, and in the upgrade script, they wrote an SQL query to handle this case. https://github.com/odoo/upgrade/pull/5496 Could you take a look at the upgrade PR and the related community one? I think a version of that query will be better!